workflow-templates/pr-name-check.yml

```
@@ -0,0 +1,28 @@
+name: 'PR name check'
+on:
+  pull_request:
+    types:
+      # Check title when opened.
+      - opened
+      # Check title when new commits are pushed.
+      # Required to use as a status check.
+      - synchronize
+
+jobs:
+  publish:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v1
+      - uses: dylanvann/check-pull-request-title@v1
+        with:
+          # Match pull request titles in the form of:
+          # Release/s123 
+          # dev => canary
+          # canary => dev
+          # canary => master
+          # master => dev
+          # MOBILE-123123 - fix bug
+          # MOBILE-123123 MOBILE-34534 - fix bug
+          # hotfix/MOBILE-123123 - fix sueprbug
+          # Revert MOBILE-123123 - revert because Lu didn't like it
+          pattern: '^((((hotfix\/|Revert\s|fix:\s|feat:\s|perf:\s|)(MOBILE|IL|NYC|CORE|DS)-\d*\s)+-\s.+)|Release\/s\d+|canary => dev|canary => master|master => dev|dev => canary)'
```

this is some serious regex.
```suggestion
          pattern: '^((dev|canary|master|main) => dev)|Release\/s\d+|(hotfix\/|Revert\s)?((MOBILE|IL|NYC|CORE|DS)-\d*\s)+-\s.+)
```
I think this is a little cleaner, if we make the assumption that `branch => branch` types of PRs should only be done merged into dev, which i think should be the case.

Also I think you need to escape `/` but maybe thats wrong